src/generalizedlinearmixedmodel.jl

```diff
@@ -426,6 +426,7 @@ function GeneralizedLinearMixedModel(
             LMM.optsum,
         )
     end
+    X = view(LMM.X, :, 1:(LMM.feterm.rank))
```

Comment:
I believe this is the same as
```suggestion
    X = fullrankx(LMM.feterm)
```